src/docs/sphinx/quickstart_guide/zero_to_axom.rst

```diff
@@ -7,23 +7,94 @@
+The easiest path to build Axom's dependencies is to use `Spack <https://github.com/spack/spack>`_.
+This has been encapsulated using `Uberenv <https://github.com/LLNL/uberenv>`_. Uberenv helps by
+doing the following:
 
+* Pulls a blessed version of Spack locally
+* If you are on a known operating system (like TOSS3), we have defined Spack configuration files
+  to keep Spack from building the world
+* Installs our Spack packages into the local Spack
+* Simplifies whole dependency build into one command
 
-.. when on github   git clone --recursive https://github.com/llnl/axom.git?
+Uberenv will create a directory containing a Spack instance with the required Axom
+dependencies installed.
 
-.. code:: bash
+It also generates a host-config file (``<config_dependent_name>.cmake``)
+at the root of Axom repository. This host-config defines all the required information for building
+Axom.
 
-    $ git clone --recursive git@github.com:LLNL/axom.git
-    $ cd axom
-    $ python scripts/uberenv/uberenv.py --install --prefix="build"
+.. code-block:: bash
 
+   $ python3 scripts/uberenv/uberenv.py
 
-After this completes, ``build/axom-install`` will contain an Axom install.
+.. note::
+  On LC machines, it is good practice to do the build step in parallel on a compute node.
+  Here is an example command: ``salloc -ppdebug -N1-1 python3 scripts/uberenv/uberenv.py``
+
+Unless otherwise specified, Spack will default to a compiler.  This is generally not a good idea when
+developing large codes. To specify which compiler to use, add the compiler specification to the ``--spec`` Uberenv
+command line option. Supported compiler specs can be found in the Spack compiler files in our repository:
+``scripts/spack/configs/<platform>/compilers.yaml``.
+
+We currently regularly test the following Spack configuration files:
+
+* Linux Ubuntu 20.04 (via Windows WSL 2)
+* TOSS 3 (On ruby at LC)
+* BlueOS (On Lassen at LC)
+
+To install Axom on a new platform, it is a good idea to start with a known Spack configuration directory
+(located in the Axom repo at ``scripts/spack/configs/<platform>``). The ``compilers.yaml`` file
+describes the compilers and associated flags required for the platform and the ``packages.yaml`` file
+describes the low-level libraries on the system to prevent Spack from building the world. Documentation on
+these configuration files is located in the `Spack docs <https://spack.readthedocs.io/en/latest/configuration.html>`_.
+
+Some helpful uberenv options include :
+
+* ``--spec=+cuda`` (build Axom with CUDA support)
+* ``--spec=+devtools`` (also build the devtools with one command)
+* ``--spec=%clang@10.0.0`` (build with a specific compiler as defined in the ``compiler.yaml`` file)
+* ``--spack-config-dir=<Path to spack configuration directory>`` (use specific Spack configuration files)
+* ``--prefix=<Path>`` (required, build and install the dependencies in a particular location)
+
```

Comment:
It might be good to also provide a link to the uberenv docs.
E.g. one of these: 
* https://uberenv.readthedocs.io/en/latest/ 
* https://uberenv.readthedocs.io/en/latest/#command-line-options